### DISCUSSION TITLE
Does Prism fully support AOT?

### DISCUSSION THREAD

**Discussion by JusterZhu** [PrismLibrary/Prism#3183] [WPF]
Do Prism-wpf and Prism-avalonia perfectly support AOT compilation when using DryIoC for reflection-based plugin loading, or to what extent is it supported? Are there any drawbacks? Are there differences between the Windows and Linux platforms?

**Comment by JusterZhu** [comment]
@dansiegel @brianlagunas help

**Reply by DamianSuess** [reply]
@JusterZhu,

As the publisher of Prism.Avalonia, this feature is still under evaluation.

Though you may have some success with smaller applications, we cannot assure 100% success across the various Linux distributions, Win, Mac, etc.

**Reply by JusterZhu** [reply]
thanks

**Comment by DamianSuess** [comment]
@JusterZhu The following is an example of creating a release package which is self-contained. This has been tested with Prism.Avalonia.

Check out the [Runtime Identifier Catalog](https://learn.microsoft.com/en-us/dotnet/core/rid-catalog) for additional runtimes, and [DotNet Publish](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-publish) for more info. This can be useful when building in AOT does not work out for your (larger) applications.

```sh
# Windows x64
dotnet publish source/MyProject/MyProject.csproj -c Release -r win-x64 -o publish/MyProject/win-x64 -p:PublishReadyToRun=true -p:PublishSingleFile=true

# Raspberry PI - Linux ARM-x64
dotnet publish source/MyProject/MyProject.csproj -c Release -r linux-x64 -o publish/MyProject/linux-arm64 -p:PublishReadyToRun=true -p:PublishSingleFile=true
```
